### ISSUE TITLE
Would OpenSSL consider publishing vulnerabilities in the OSV format as well as CVE?

### ISSUE THREAD

**Issue by andrewpollock** [openssl/openssl#25497]
Would OpenSSL be open (no pun intended) to exploring publishing vulnerability records in the [OSV format](https://github.com/ossf/osv-schema) as well (including the back catalogue of CVEs since 2016)?

Due to its more [unique versioning scheme](https://wiki.openssl.org/index.php/Versioning), [OpenSSL is a little problematic for OSV.dev's current normalization code to reason about](https://github.com/google/osv.dev/issues/2220). 

There's only 119 CVEs that I can see issued by OpenSSL's CNA since 2016, so the raw count isn't particularly significant, but I think given the significance of OpenSSL as a piece of open source C-based software, there's an outsized positive impact for vulnerability detection with OSV.dev's data, API and tooling if the vulnerability data is available.

Background:
- https://osv.dev/blog/posts/introducing-broad-c-c++-support/
- http://osv.dev/faq

**Comment by andrewpollock** [comment]
OSV.dev's onboarding process is a little bit bespoke and toilsome at the moment, but something we're continuously improving on and streamlining with each new data source onboarded. I would like to get it to the point of being much more checklist/cookbook driven than it currently is. This is an (ongoing) experiment at further process improvement and seeks to address past feedback. Your actionable feedback is also very welcome.

In a nutshell:

- [ ] Decide if you're going to publish records via a Git repository, GCS bucket or [REST endpoint](https://google.github.io/osv.dev/rest-api-contribution/) (I'm going to assume a Git repository?)
- [ ] Create a PR to [reserve a prefix in the OSV-Schema](https://ossf.github.io/osv-schema/#id-modified-fields) (worked examples: https://github.com/ossf/osv-schema/pull/235 https://github.com/ossf/osv-schema/pull/223 https://github.com/ossf/osv-schema/pull/219)
  - [ ] We review the records you start publishing for OSV Schema [correctness](https://github.com/ossf/osv-schema/tree/main/validation) and [quality](https://google.github.io/osv.dev/data_quality.html) (the work happening under the [OSV Data Quality Program](https://github.com/orgs/google/projects/62) is also relevant here, as an FYI) as part of reviewing and merging that PR
- [ ] Create a PR to extend [purl_helpers.py](https://github.com/google/osv.dev/blob/master/osv/purl_helpers.py) (if appropriate)
- [ ] Create a PR to start [importing the records you are publishing into our test instance of OSV.dev](https://github.com/google/osv.dev/blob/master/source_test.yaml) and validate everything is working as intended there (worked example: https://github.com/google/osv.dev/pull/2086)
- [ ] Create a PR to start [importing the records you are publishing into our production environment](https://github.com/google/osv.dev/blob/master/source.yaml) (worked example: https://github.com/google/osv.dev/pull/2105)

**Known onboarding rough edges**:
- the format of the `source{,_test}.yaml` files (hopefully the example PRs plus other existing entries will make this reasonably self-evident). Specifically, FYI, the value for `type` corresponds with those defined at https://github.com/google/osv.dev/blob/381f459de12e181447731beee9ba4b06a513c586/osv/models.py#L783-L787

**Comment by t8m** [comment]
This looks like a fairly involved process, unfortunately I do not think this is likely going to be a team priority for the near (and not so near) future as the team resources are fairly limited and there are many other priorities.

So I am sorry, but unless this work is externally contributed, it is not going to happen soon.

Please note we publish the vulnerability data in parsable .json format here https://openssl-library.org/news/secjson/

**Comment by andrewpollock** [comment]
> This looks like a fairly involved process

I'd like to unpack that perception with you, because that's obviously a (perceived) barrier to adoption, and something we'd like to have be as minimal as possible. Your actionable feedback would be greatly appreciated.

Here's my take on some very ballpark time expectations, acknowledging my curse of knowledge:

- Deciding the record publishing mechanism
  - This is difficult to generalise, as it's somewhat project specific. This could take 5 minutes or it could take days, depending on the project
- Creating a PR to reserve a prefix in the OSV Schema
  - If that takes more than 30 minutes for someone with an existing PR to cargo-cult from, I'd really like to understand *why*
- Creating a PR to extend `purl_helpers.py`
  - That's something that we can assist with as necessary. For attribution and a base-level of understanding, we prefer the implementing project to have some skin the game, but it's not a hard requirement. This would depend on familiarity with Python and Purls, and there are previous PRs to crib from. I'd make a guesstimate of less than an hour in the optimal case.
- Creating a PR to start importing records into OSV.dev's Staging environment
  - I'd hope < 30 minutes, given the amount of prior art
- Creating a PR to start importing records into OSV.dev's Production environment
  - I'd hope < 5 minutes based on the previous PR

What I'd say we do lack any perspective on is the time to become productive with the OSV.dev GitHub repo for a newcomer...
- working through the new contributor's guide

@jasinner as someone who most recently walked this path, could you offer your opinion on this?

/cc @oliverchang for visibility

**Comment by t8m** [comment]
It also means to modify all our processes to also include OSV processing when we do CVE processing. This has by no means negligible impact. Our current process for doing security releases is unfortunately fairly manual and until it is better automated I do not think this is going to happen.

**Comment by andrewpollock** [comment]
Correct, there would be process-level work to handle OSV vulnerabilities. 

We have plans to create open source tooling to translate from OSV to CVE 5, so there can be a single source of truth and minimal duplicated or additional effort to vulnerability disclosure processes for OSS projects.

Is your current process publicly documented anywhere?

**Comment by t8m** [comment]
https://github.com/openssl/tools/blob/master/HOWTO-handle-security-issue.md is the current process.

**Comment by andrewpollock** [comment]
You may like to consider https://bughunters.google.com/open-source-security/security-grants if additional funding could assist?

**Comment by t8m** [comment]
We are not going to implement this.
